python-sdk/langwatch/tracer.py

```diff
@@ -1163,33 +1177,37 @@ def send_spans(
         data["spans"] = data["spans"][:256]
 
     # TODO: replace this with httpx, don't forget the custom SerializableWithStringFallback encoder
-    response = requests.post(
-        f"{langwatch.endpoint}/api/collector{force_sync}",
-        data=json.dumps(data, cls=SerializableWithStringFallback),
-        headers={
-            "X-Auth-Token": str(api_key),
-            "Content-Type": "application/json",
-        },
-    )
 
-    if response.status_code == 429:
-        json = response.json()
-        if "message" in json and "ERR_PLAN_LIMIT" in json["message"]:
-            warn(json["message"])
+    if langwatch.enabled and random() < langwatch.sampling_rate and not disable_sending:
```

Comment:
You could do an early return here like the api key above instead of indenting everything on the if